Source/Task/WaitTimer_stl.cpp

```diff
@@ -121,6 +134,9 @@ void TimerQueue::Worker() noexcept
     std::unique_lock<std::mutex> lock{ m_mutex };
     while (true)
     {
+        if (m_exitThread)
+            return;
```

Comment:
I think as written there is a race condition: if m_exitThread is set to true and the condition variable is signalled while we are looping over the queue, we will never exit

I suggest the following change:
```c++
while (!m_exitThread)
{
    // do work
    if (!m_exitThread)
    {
        // wait on condition variable
    }
}